react/src/components/AutoScalingRuleEditorModal.tsx

```diff
@@ -0,0 +1,320 @@
+import BAIModal, { BAIModalProps } from './BAIModal';
+import Flex from './Flex';
+import InputNumberWithSlider from './InputNumberWithSlider';
+import { AutoScalingRuleEditorModalCreateMutation } from './__generated__/AutoScalingRuleEditorModalCreateMutation.graphql';
+import { AutoScalingRuleEditorModalFragment$key } from './__generated__/AutoScalingRuleEditorModalFragment.graphql';
```

Comment:
You can use `AutoScalingMetricComparator` type and `AutoScalingMetricSource` type generate type based on graphql schema.
```suggestion
import { AutoScalingMetricComparator, AutoScalingMetricSource, AutoScalingRuleEditorModalCreateMutation } from './__generated__/AutoScalingRuleEditorModalCreateMutation.graphql';
```